packages/console/src/components/OrganizationRolePermissionsAssignmentModal/use-organization-scopes-assignment.ts

```diff
@@ -0,0 +1,25 @@
+import { type OrganizationScope } from '@logto/schemas';
+import { useMemo, useState } from 'react';
+import useSWR from 'swr';
+
+function useOrganizationScopesAssignment(assignedScopes: OrganizationScope[] = []) {
+  const [selectedData, setSelectedData] = useState<OrganizationScope[]>([]);
+
+  const { data: organizationScopes } = useSWR<OrganizationScope[]>('api/organization-scopes');
+
+  const availableDataList = useMemo(
+    () =>
+      (organizationScopes ?? []).filter(
+        ({ id }) => !assignedScopes.some((scope) => scope.id === id)
+      ),
+    [organizationScopes, assignedScopes]
+  );
+
+  return {
+    selectedData,
+    setSelectedData,
+    availableDataList,
+  };
```

Comment:
I think this return value is a bit complicated, might need to wrap in another `useMemo` instead